src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/Command/RevokeOwnSecondFactorCommand.php

```diff
@@ -19,18 +19,20 @@
 namespace Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Command;
 
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Command\AbstractCommand;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Command\RaExecutable;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\Command\SelfServiceExecutable;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * May be executed by a Registrant to revoke one of his/her own verified or vetted second factors.
  */
-class RevokeOwnSecondFactorCommand extends AbstractCommand
+class RevokeOwnSecondFactorCommand extends AbstractCommand implements SelfServiceExecutable, RaExecutable
```

Comment:
Afaik this command isn't executed from the RA application, or do you intend to execute this when an RA removes his/her own second factor in the RA?